rust/types/src/strategies.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{Operation, OperationRecord, ScalarEncoding, UpdateMetadata, UpdateMetadataValue};
+use proptest::prelude::*;
+
+/**
+ * Strategy for metadata.
+ */
+
+fn arbitrary_update_metadata(
+    min_pairs: usize,
+    max_pairs: usize,
+) -> impl Strategy<Value = UpdateMetadata> {
+    let num_pairs = (min_pairs..=max_pairs).boxed();
+
+    num_pairs
+        .clone()
+        .prop_flat_map(|num_pairs| {
+            let keys = proptest::collection::vec(proptest::arbitrary::any::<String>(), num_pairs);
+
+            let values = proptest::collection::vec(
+                prop_oneof![
+                    proptest::strategy::Just(UpdateMetadataValue::None),
+                    proptest::bool::ANY.prop_map(|v| UpdateMetadataValue::Bool(v)),
+                    proptest::arbitrary::any::<i64>().prop_map(|v| UpdateMetadataValue::Int(v)),
+                    (-1e6..1e6f64).prop_map(|v| UpdateMetadataValue::Float(v)),
+                    proptest::arbitrary::any::<String>()
+                        .prop_map(|v| { UpdateMetadataValue::Str(v) }),
+                ],
+                num_pairs,
+            );
+
+            (keys, values)
+        })
+        .prop_map(|(keys, values)| keys.into_iter().zip(values).collect::<UpdateMetadata>())
+}
+
+/**
+ * Strategy for operation record.
+ */
+
+impl Arbitrary for OperationRecord {
```

Comment:
this was originally much more complex and generating a Chunk/Vec of `OperationRecord`s, but I realized this is fine for now